envoy/http/codec.h

```diff
@@ -220,7 +220,7 @@ class RequestDecoder : public virtual StreamDecoder {
    */
   virtual void sendLocalReply(Code code, absl::string_view body,
                               const std::function<void(ResponseHeaderMap& headers)>& modify_headers,
-                              const absl::optional<Grpc::Status::GrpcStatus> grpc_status,
+                              Grpc::Status::LocalReplyGrpcStatusOptionPtr grpc_status,
```

Comment:
How about use absl::optional<LocalReplyGrpcStatusOption> in the function argument? 

In the struct LocalReplyGrpcStatusOption, not to use absl::optional

Reply:
As mentioned in the description, later we will add [gRPC status](https://github.com/envoyproxy/envoy/pull/20620#discussion_r842742436) in struct for error details , whose copy is not trivial. Use absl::optional will perform copy so prefer using unique_ptr at the beginning.